core/iwasm/libraries/lib-socket/inc/wasi_socket_ext.h

```diff
@@ -445,6 +445,62 @@ __wasi_sock_set_send_buf_size(__wasi_fd_t fd)
         __imported_wasi_snapshot_preview1_sock_set_send_buf_size((int32_t)fd);
 }
 
+int32_t
+__imported_wasi_snapshot_preview1_sock_get_recv_timeout(int32_t arg0,
+                                                        int32_t arg1)
+    __attribute__((__import_module__("wasi_snapshot_preview1"),
+                   __import_name__("sock_get_recv_timeout")));
+
+static inline __wasi_errno_t
+__wasi_sock_get_recv_timeout(__wasi_fd_t fd, uint64_t *timeout_us)
+{
+    return (__wasi_errno_t)
+        __imported_wasi_snapshot_preview1_sock_get_recv_timeout(
+            (int32_t)fd, (int32_t)timeout_us);
+}
+
+int32_t
+__imported_wasi_snapshot_preview1_sock_set_recv_timeout(int32_t arg0,
+                                                        int32_t arg1)
+    __attribute__((__import_module__("wasi_snapshot_preview1"),
+                   __import_name__("sock_set_recv_timeout")));
+
+static inline __wasi_errno_t
+__wasi_sock_set_recv_timeout(__wasi_fd_t fd, uint64_t *timeout_us)
+{
+    return (__wasi_errno_t)
+        __imported_wasi_snapshot_preview1_sock_set_recv_timeout(
+            (int32_t)fd, (int32_t)timeout_us);
```

Comment:
Here I'm converting the `timeout_us` pointer to `int32_t`, rather than passing the value as int64. It's because of this limitation:

> export_native_api.md - As function parameters are always passed in 32 bits numbers, you can also use 'i' for the pointer type argument, then you must do all the address conversion and boundary checking in your native function.

Tested that passing ULLONG_MAX (rounded down to nearest 1000) sets and receives properly. There should probably be a specific test case for this though

Reply:
Doesn't that refer to pointer types? I think it's possible to pass int64, that's why we have `I` (uppercase `i`) as a registration parameter. You can also see some examples in wasi libc, e.g. `fd_pwrite`'s `offset` parameter is of `wasi_filesize_t` type which is a type alias of `uint64_t`.

Reply:
Yeah you're right I can use `I`, misunderstood this part. Will update